runtime/compiler/env/j9method.cpp

```diff
@@ -5839,6 +5840,44 @@ TR_ResolvedJ9Method::longConstant(I_32 cpIndex)
    return *((uint64_t *) & romLiterals()[cpIndex]);
    }
 
+void *
+TR_ResolvedJ9Method::getConstantDynamicTypeFromCP(I_32 cpIndex)
+   {
+   return jitGetConstantDynamicTypeFromCP(fej9()->vmThread(), cp(), cpIndex);
+   }
+
+bool
+TR_ResolvedJ9Method::isConstantDynamic(I_32 cpIndex)
+   {
+   TR_ASSERT(cpIndex != -1, "cpIndex shouldn't be -1");
+   UDATA cpType = J9_CP_TYPE(J9ROMCLASS_CPSHAPEDESCRIPTION(cp()->ramClass->romClass), cpIndex);
+   return (J9CPTYPE_CONSTANT_DYNAMIC == cpType);
+   }
+
+// If first slot is non null, the CP entry is resolved to a non-null value.
+// Else if second slot is Void.class, the CP entry is resolved to null (0) value.
+// Other casese, the CP entry is considered unresolved.
+bool
+TR_ResolvedJ9Method::isUnresolvedConstantDynamic(I_32 cpIndex)
```

Comment:
Does the first slot represent the value of the condy and the second slot represent the resolved class of the condy? Maybe it's better to say this explicitly in the comment to avoid confusion especially when the name of the second slot is called `exception`.

Reply:
Both value and exception slots are of object pointer type. If first slot is non null, the CP entry is resolved to a non-null value. Else if second slot is the class object of j/l/Void, the CP entry is resolved to null (0) value. We retrieve the Void class object via javaVM->voidReflectClass->classObject, which is protected by VMAccessCrtiticalSection to ensure vm access. Other cases, the CP entry is considered unresolved. Added VMAccessCrtiticalSection code and more comments.